staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth.go

```diff
@@ -36,25 +36,25 @@ var (
 // AzureAuthConfig holds auth related part of cloud config
 type AzureAuthConfig struct {
 	// The cloud environment identifier. Takes values from https://github.com/Azure/go-autorest/blob/ec5f4903f77ed9927ac95b19ab8e44ada64c1356/autorest/azure/environments.go#L13
-	Cloud string `json:"cloud" yaml:"cloud"`
+	Cloud string `json:"cloud,omitempty" yaml:"cloud,omitempty"`
```

Comment:
What's the reason for omitempty changes?

Reply:
(makes sense to me, just wondering why)

Reply:
it's because of merging configs from secret. Without omitempty, all fileds would get an empty value if it's not set. With omitempty, this won't happen.

Reply:
Is this rollback safe?

Reply:
yep, there're default values being set after parsing the config.